docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+It has two subclasses: `Todo` and `Deadline`. `Deadline` items also have the date information and countdown for the deadline date, which are not included in `Todo` tasks. 
+
+#### Event
+
+The `Event` class stores the information of all `Event` items, including date, time, venue, status whether it is overdue, and any other information if added. 
+It has two subclasses: SchoolEvent and Activity. `Activity` items can have other details.
+All `SchoolEvent` items must have a module code.  
+Furthermore, the `SchoolEvent` class has four subclasses which are `Lecture`, `Tutorial`, `Lab`, and `Exam`. Among them only `Exam` items have a countdown for the exam date. 
+All `Event` items update the information or provide needed information about `Event` items when executing commands related to events or saving the information to `Storage`.
+
+Design consideration: `SchoolEvent` is modelled after NUS modules to cater to our intended users.
+
+## Implementation
+This section describes the details of how each feature is implemented.
+
+### Add a calendar item feature
+This feature is facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/add_command_SD.JPG" alt="" width="750"/><br/>
+
+Note: A self-call to a specific method to add the new `CalendarItem` to the `CalendarList` is done based on the `CalendarItem` being added.  
+
+`CalendarItems` with deviations to sequence diagram above: 
+1. All `CalendarItems` with a date or time attribute calls the `TimeParser` class to interpret the date and time input of the user. 
+2. `Lecture`, `Lab`, `Tutorial`, or `Exam` type does a self-call for an additional check using the function `isValid(command)`. It ensures the module code included in the item is valid.
+3. `Lecture`, `Lab`, or `Tutorial`can be added as recurring items (multiple addition of the same class).   
+
+### Mark a task as done feature
+This feature is facilitated by `DoneCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/done_command_SD.JPG" alt="" width="750"/><br/>
+Note: The command first extracts the task number from the user input prior to `convertTaskNumberToCalendarNumber`. 
+There will be a check in the function `markTaskAsDone(calendarNumber)` to ensure that the calendar item being marked as done is a task. 
+
+
+### Additional information of an event feature
+#### Add additional information
+This feature is facilitated by `AddInfoCommand`.
+The following sequence diagram show how the `execute()` operation of `AddInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/addInfoCommand_SD.JPG" alt="" width="750"/><br/>
```

Comment:
![image](https://user-images.githubusercontent.com/18654251/97659672-826ee400-1aaa-11eb-8fa7-5e3d7632c83f.png)

There seems to an inconsistent format with :CalendarList having a dotted line